reframe/frontend/cli.py

```diff
@@ -595,14 +595,25 @@ def main():
     if options.restore_session is not None:
         # We need to load the failed checks only from a report
         if options.restore_session:
-            filename = options.restore_session
+            filenames = options.restore_session.split(',')
         else:
-            filename = runreport.next_report_filename(
+            filenames = [runreport.next_report_filename(
                 osext.expandvars(site_config.get('general/0/report_file')),
                 new=False
-            )
+            )]
+
+
+        reports = []
+        for filename in filenames:
+            report = runreport.load_report(filename)
+            reports.append(report)
+
+        # pop the last report, merge in the other reports
+        # this is almost certainly not the right way to go about this
+        report = reports.pop()
+        for rpt in reports:
+            report._cases_index.update(rpt._cases_index)
```

Comment:
Instead of merging the reports, I am going to use the other reports as fallback in case a test case is not found in the primary one.